### PR TITLE
Implement sync-collection REPORT on home-set collections

### PR DESCRIPTION
**Pull request by ksmurchison**
Tested by hand but needs Cass tests

### CODE REVIEW

File: imap/http_caldav.c [hunk 1]
```diff
@@ -610,7 +610,12 @@ static struct meth_params caldav_params = {
       { NS_CALDAV, "calendar-data", &caldav_import } },
     { CALDAV_SUPP_DATA, &caldav_put },
     { 0, caldav_props },                        /* Allow infinite depth */
-    caldav_reports
+    caldav_reports,
+    { offsetof(struct mboxname_counters, caldavmodseq),
```
Comment (brong) [inline imap/http_caldav.c:614]: This is really clever and I hope portable!  It probably needs a comment somewhere explaining how and why it's used,.

File: imap/http_dav.c [hunk 2]
```diff
@@ -7524,6 +7531,110 @@ static int updates_cb(void *rock, void *data)
...
+        len++;
+    }
+
+    mbname_t *mbname = mbname_from_intname(mbentry->name);
+    snprintf(p, MAX_MAILBOX_PATH - len, "%s/",
+             strarray_nth(mbname_boxes(mbname), -1));
+    mbname_free(&mbname);
+
+    /* Skip special calendar collections */
+    if (fctx->req_tgt->namespace->id == URL_NS_CALENDAR &&
```
Comment (brong) [inline imap/http_dav.c:7567]: It would be nice to have a flag on these rather than exact name matches, but we have what we have for now!

**Review by brong — APPROVED** [review]
This looks fine, it could do with some more comments the modseq pair stuff, but I can understand it all.

And yeah, tests would be nice - particularly tests that it doesn't leak collections that it shouldn't (including ACL checks)

File: imap/http_caldav.c (hunk 1 above)
Comment (ksmurchison) [reply imap/http_caldav.c:614]: libical uses stuff like this, so I assume its portable

File: imap/http_dav.c (hunk 2 above)
Comment (ksmurchison) [reply imap/http_dav.c:7567]: Agreed.  We have flags when parsing a method URL, but not when we're iterating over mailboxes.

**Comment by ksmurchison** [comment]
@brong Added a Cass test

**Review by brong — APPROVED** [review]
Thanks for the test!  Looks good

**Pull request merged**